src/main/java/com/o19s/es/ltr/feature/PrebuiltFeatureSet.java

```diff
@@ -35,34 +37,46 @@ public String name() {
     }
 
     @Override
-    public List<Feature> features() {
-        return Arrays.asList(features);
+    public PrebuiltFeature[] asArray() {
+        return features.toArray(new PrebuiltFeature[features.size()]);
```

Comment:
I really don't know yet how it'll look like for other implementations...
This class has been created to start using the new model with the existing query builder. 
This will likely change in the future.